src/main/java/org/hawkular/client/android/backend/BackendClient.java

```diff
@@ -255,26 +258,45 @@ public void getMetrics(@NonNull Environment environment, @NonNull Resource resou
         readPipe(BackendPipes.Names.METRICS, uri, callback);
     }
 
-    public void getMetricDataAvailability(@NonNull Resource resource,
+    public void getMetricDataAvailability(@NonNull Metric metric, long bucket,
                                           @NonNull Date startTime, @NonNull Date finishTime,
-                                          @NonNull Callback<List<MetricData>> callback) {
+                                          @NonNull Callback<List<MetricAvailabilityBucket>> callback) {
         Map<String, String> parameters = new HashMap<>();
         parameters.put(BackendPipes.Parameters.START, String.valueOf(startTime.getTime()));
         parameters.put(BackendPipes.Parameters.FINISH, String.valueOf(finishTime.getTime()));
+        parameters.put(BackendPipes.Parameters.BUCKETS, String.valueOf(bucket));
 
-        URI uri = Uris.getUri(String.format(BackendPipes.Paths.METRIC_DATA_AVAILABILITY, resource.getId()), parameters);
+        URI uri = Uris.getUri(String.format(BackendPipes.Paths.METRIC_DATA_AVAILABILITY,
+                Uris.getEncodedParameter(metric.getId())), parameters);
 
         readPipe(BackendPipes.Names.METRIC_DATA_AVAILABILITY, uri, callback);
     }
 
-    public void getMetricDataGauge(@NonNull Metric metric,
+    public void getMetricDataCounter(@NonNull Metric metric, long bucket,
```

Comment:
I think you should try to not duplicate this via copy&paste, but have one more generic method `getMetricData` where you pass in the type as parameter. It is a bit more tricky for availability above, but there you could at least extract the setting of start/end/buckets into a helper method